api/backend/app/controllers/agent_router.py

```diff
@@ -41,8 +43,8 @@ async def get_agent(self, agent_id: str):
         return agent
 
     @put("/agents/{agent_id}", status_code=HTTPStatus.OK)
-    async def update_agent(self, agent_id: str, agent_data: AgentUpdateDTO):
-        updated_agent = await self.agent_service.update_agent(agent_id, agent_data)
+    async def update_agent(self, agent_id: str, agent_data: AgentUpdateDTO, user: dict = Depends(verify_cookie)):
```

Comment:
use TypedDict for user

Reply:
Created Typed Dict Model for User and replaced previous dict accordingly.